torch/csrc/api/src/optim/rmsprop.cpp

```diff
@@ -11,55 +11,173 @@
+      }
+
+      square_avg.mul_(alpha).addcmul_(grad, grad, 1 - alpha);
+
+      Tensor avg;
+      if (options.centered()) {
+        auto& grad_avg = state.grad_avg();
+        grad_avg.mul_(alpha).add_(grad, 1-alpha);
+        avg = square_avg.addcmul(grad_avg, grad_avg, -1).sqrt_().add_(options.eps());
+      } else {
+        avg = square_avg.sqrt().add_(options.eps());
+      }
+
+      if (options.momentum() > 0) {
+        auto& buf = state.momentum_buffer();
+        buf.mul_(options.momentum()).addcdiv_(grad, avg);
+        // Need to avoid version tracking for parameter.
+        p.data().add_(buf, -options.lr());
+      } else {
+        // Need to avoid version tracking for parameter.
+        p.data().addcdiv_(grad, avg, -options.lr());
+      }
     }
   }
 }
 
+void RMSprop::add_parameters(const std::vector<Tensor>& parameters) {
+  return _add_parameters_new_design(parameters);
+}
+
+const std::vector<Tensor>& RMSprop::parameters() const noexcept {
+  return _parameters_new_design();
+}
+
+std::vector<Tensor>& RMSprop::parameters() noexcept {
+  return _parameters_new_design();
+}
+
+size_t RMSprop::size() const noexcept {
+  return _size_new_design();
+}
+
 void RMSprop::save(serialize::OutputArchive& archive) const {
   serialize(*this, archive);
 }
 
 void RMSprop::load(serialize::InputArchive& archive) {
-  serialize(*this, archive);
+  IValue pytorch_version;
+  if (archive.try_read("pytorch_version", pytorch_version)) {
+    serialize(*this, archive);
+  }
+  else { // deserializing archives saved in old format (prior to version 1.5.0)
+    TORCH_WARN(
+      "Your serialized RMSprop optimizer is still using the old serialization format. "
+      "The step value in state will be set to 0 because the old RMSprop optimizer didn't track the step value."
+      "You should re-save your RMSprop optimizer to use the new serialization format.");
+    std::vector<Tensor> square_average_buffers;
+    std::vector<Tensor> momentum_buffers;
+    std::vector<Tensor> grad_average_buffers;
+    torch::optim::serialize(archive, "square_average_buffers", square_average_buffers);
+    torch::optim::serialize(archive, "momentum_buffers", momentum_buffers);
+    torch::optim::serialize(archive, "grad_average_buffers", grad_average_buffers);
+    // since there were no param_groups prior to version 1.5.0, assuming all tensors are now in one param_group
+    std::vector<Tensor> params = param_groups_.at(0).params();
+    for (size_t idx = 0; idx < square_average_buffers.size(); idx++) {
+      auto state = std::make_unique<RMSpropParamState>();
+      state->square_avg(square_average_buffers[idx]);
+      if(idx < momentum_buffers.size()) {
+        state->momentum_buffer(momentum_buffers.at(idx));
+      }
+      if(idx < grad_average_buffers.size()) {
+        state->grad_avg(grad_average_buffers.at(idx));
+      }
+      state_[c10::guts::to_string(params[idx].unsafeGetTensorImpl())] = std::move(state);
```

Comment:
should we explicitly set `state->step(0)` here? The default value of `step` in state might not be 0 (source: https://stackoverflow.com/a/16782131)

Reply:
fair will do